rir/src/interpreter/interp.cpp

```diff
@@ -1472,25 +1472,17 @@ SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP env,
     assert(c->info.magic == CODE_MAGIC);
     bool existingLocals = localsBase;
 
-    bool smallCache = c->bindingsCount < MAX_CACHE_SIZE;
-    size_t cacheSize = smallCache ? c->bindingsCount + 1 : MAX_CACHE_SIZE;
-    if (cacheSize == 1)
-        cacheSize = 0;
     BindingCache* bindingCache;
     if (cache) {
         bindingCache = cache;
     } else {
-        bindingCache =
-            (BindingCache*)(alloca(sizeof(BindingCache) * cacheSize));
-        /*
-         * In case we are entering the executing of a PIR optimized function
-         * we delay the cache cleaning until we know we are actually requiring
-         * an environment (mk_env).
-         */
-        if (env != symbol::delayedEnv)
```

Comment:
This saves one cache clearing in every pir function activation. Why remove it?

Reply:
because it is faster (see my comment)

Reply:
my guess is clearing right after alloca is faster than later

Reply:
This is a sensible guess. But the thing is that we are not cleaning now and not cleaning later. We are doing both. The only difference is when a PIR function, before creating an env changes the environment. Maybe we can just do something with those functions so that we do the cleaning at startup.

Reply:
I know that we are cleaning twice. was still faster.